lib/pigeon/config_error.ex

```diff
@@ -0,0 +1,14 @@
+defmodule Pigeon.ConfigError do
+  defexception reason: nil, config: nil
+
+  @impl true
+  def message(%{config: config, reason: reason}) do
+    """
+    #{reason}
+
+    The following configuration was given: 
+
+    #{inspect(config, pretty: true)}
```

Comment:
If the `key` or `config` are provided (rather than `keyfile`/`configfile`), inspecting these may leak one or both of them to logs for other configuration errors. I would rather see security values stripped. Based on my reading of the code prior to this, the filename provided is removed (changed to `nil`) which doesn’t necessarily help debug what the bad filename is.

Reply:
This _might_ be resolvable with a `@derive Inspect`, but as I’ve not done that with anything I need special inspection on, I can’t rightly say the best way to do that.

Reply:
Fair point, I'll look up how phoenix handles it.